scripts/rff/historical_weather_variability/4km/preprocess_4km.py

```diff
@@ -0,0 +1,98 @@
+import csv
+import glob
+import numpy as np
+import os
+from osgeo import gdal
+import requests
+import sys
+
+SCRIPTS_DIR = os.path.dirname(os.path.dirname(
+    os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+sys.path.append(SCRIPTS_DIR)
+from rff import util
+
+bandname_to_gdcStatVars = {
+    "std_dev": "StandardDeviation_<c_var>",
+    "skewness": "Skewness_<c_var>",
+    "kurtosis": "Kurtosis_<c_var>",
+    "hpi": "HeavyPrecipitationIndex",
+    "cdd": "ConsecutiveDryDays"
+}
+
+def get_grid_latlon(ds, x, y):
+    xmin, xres, _, ymax, _, yres = ds.GetGeoTransform()
+    return ymax + y*yres, xmin + xres*x
+
+def create_4km_grids(output_csv, sample_gtiff):
+    sample_ds = gdal.Open(sample_gtiff)
+    raster = sample_ds.GetRasterBand(1).ReadAsArray()
+    shape = raster.shape
+    headers = ['latitude', 'longitude', 'typeOf', 'dcid', 'containedInPlace']
+    with open(output_csv, 'w', newline='') as csvfile:
+        csvwriter = csv.writer(csvfile)
+        csvwriter.writerow(headers)
+        typeOf = "dcid:GeoGridPlace_4KM"
+        url = "https://geocoding.geo.census.gov/geocoder/geographies/coordinates?benchmark=Public_AR_Current&vintage=ACS2021_Current&format=json"
+        for y in range(shape[0]):
+            for x in range(shape[1]):
+                if not np.isnan(raster[y][x]):
+                    lat, lon = get_grid_latlon(sample_ds, x, y)
+                    resp = requests.get(f"{url}&x={lon}&y={lat}")
+                    geo_data = resp.json()["result"]["geographies"]
+                    if geo_data != {}:
+                        geoid = f"dcid:geoId/{geo_data['Counties'][0]['GEOID']}"
+                        dcid = 'dcid:grid_4km/' + str(lat) + '_' + str(lon)
```

Comment:
We might want to format the lat/lng with fixed number of decimal places (3 or 4 digits?)